WireBST.java

```diff
@@ -149,21 +232,54 @@ public void maxWireBST(WireBSTNode node){
 
     }
 
-    // print the WireBST as pre order
+    /**
+     * print the wireBST as pre oder
+     * @param node
+     */
     public void preOrderScan(WireBSTNode node){
 
+        if(node == null) return;
+
+        System.out.println(node._studentNum + " ");             //print the node first
+        System.out.println(node._studentName + " ");
+
+        preOrderScan(node.getLeft());                           //recursive call on left subtree
+
+        preOrderScan(node.getRight());                          //recursive call on right subtree
 
     }
 
-    // print the WireBST as in order
+    /**
+     * print the wireBST as in order
+     * @param node
+     */
     public void inOrderScan(WireBSTNode node){
```

Comment:
better be like this:
WireBSTNode temp = minWireBST() ;   //  temp now poit on the minimun of the tree, which is the leftest node.
for ( ; temp; temp = temp.getLeft() ) 
     toString(temp);                           // toString will return a string the describe the node (without the wires) : " <studentNum>: <studentName> "